presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -721,16 +747,24 @@ private void revokeRole(String role, String granteeName, PrincipalType granteeTy
         }
     }
 
-    private List<String> getPrestoViews(String databaseName)
+    private List<String> doGetTablesWithParameter(String databaseName, String parameterKey, String parameterValue)
             throws TException
     {
+        checkArgument(TABLE_PARAMETER_SAFE_KEY_PATTERN.matcher(parameterKey).matches(), "The parameter key '%s' does not match the pattern '%s'", parameterKey, TABLE_PARAMETER_SAFE_KEY_PATTERN.pattern());
+        /*
+         * The parameter value is restricted to have only alphanumeric characters so that it's safe to be used against HMS.
+         * When using with a LIKE operator, the HMS may want the parameter value to follow a Java regex pattern or a SQL
+         * pattern. And it's hard to predict the HMS's behavior from outside.
+         * Also, by restricting parameter values, we avoid problem how to quote them when passing within the filter string.
+         */
+        checkArgument(TABLE_PARAMETER_SAFE_VALUE_PATTERN.matcher(parameterValue).matches(), "The parameter value '%s' does not match the pattern '%s'", parameterValue, TABLE_PARAMETER_SAFE_VALUE_PATTERN.pattern());
```

Comment:
for context -- i wanted to have the pattern here (but I'm okay if this is not included)

still, if the value can contain invalid characters (as per check), we should quote the value in the message (`'%s'`), so that any trailing offending chars remain visible